diff -r -u boost_1_34_1/tools/build/v2/tools/stage.jam boost_1_34_1.patched/tools/build/v2/tools/stage.jam
--- boost_1_34_1/tools/build/v2/tools/stage.jam	2006-11-10 09:20:05.000000000 -0800
+++ boost_1_34_1.patched/tools/build/v2/tools/stage.jam	2007-06-19 09:07:56.625000000 -0700
@@ -351,7 +351,8 @@
     
     rule run ( project name ? : property-set : source : multiple ? )
     {
-        if [ $(property-set).get <os> ] in NT CYGWIN
+        if [ $(property-set).get <os> ] in NT CYGWIN ||
+            [ $(property-set).get <target-os> ] in windows cygwin
         {
             # Relinking is never needed on NT
             return [ stage.copy-file $(project) 
@@ -383,7 +384,8 @@
     
     rule run ( project name ? : property-set : source : multiple ? )
     {                        
-        if [ $(property-set).get <os> ] = NT
+        if [ $(property-set).get <os> ] in NT CYGWIN ||
+            [ $(property-set).get <target-os> ] in windows cygwin
         {
             local copied = [ stage.copy-file $(project) 
               : $(source) : $(property-set) ] ; 
