diff -r -b -u boost_1_34_1/boost/asio/detail/posix_fd_set_adapter.hpp boost_patched/boost/asio/detail/posix_fd_set_adapter.hpp
--- boost_1_34_1/boost/asio/detail/posix_fd_set_adapter.hpp	2007-10-25 03:17:16.000000000 -0700
+++ boost_patched/boost/asio/detail/posix_fd_set_adapter.hpp	2007-10-31 16:56:01.000000000 -0700
@@ -43,7 +43,10 @@
     FD_SET(descriptor, &fd_set_);
   }
 
-  bool is_set(socket_type descriptor) const
+  bool is_set(socket_type descriptor)
+#ifndef __APPLE__
+  const
+#endif
   {
     return FD_ISSET(descriptor, &fd_set_) != 0;
   }
