diff -ru icu/source/configure icu.new/source/configure
--- icu/source/configure	2009-07-01 11:51:26.000000000 -0700
+++ icu.new/source/configure	2010-02-22 15:14:06.000000000 -0800
@@ -7062,7 +7062,7 @@
 	# there are some -arch flags.  Note that *ppc* also matches
 	# ppc64.  This check is also rather less than ideal.
 	case "${CC} ${CFLAGS} ${CPPFLAGS} ${LDFLAGS}" in  #(
-	  *-arch*ppc*|*-arch*i386*|*-arch*x86_64*) ac_cv_c_bigendian=universal;;
+	  *-arch*ppc*) ac_cv_c_bigendian=yes;;
 	esac
 else
   $as_echo "$as_me: failed program was:" >&5
diff -ru icu/source/runConfigureICU icu.new/source/runConfigureICU
--- icu/source/runConfigureICU	2009-07-01 11:51:26.000000000 -0700
+++ icu.new/source/runConfigureICU	2010-02-22 15:12:54.000000000 -0800
@@ -250,8 +250,8 @@
     MacOSX)
         THE_OS="MacOS X (Darwin)"
         THE_COMP="the GNU C++"
-        RELEASE_CFLAGS='-O2'
-        RELEASE_CXXFLAGS='-O2'
+        RELEASE_CFLAGS='-O2 -arch x86_64'
+        RELEASE_CXXFLAGS='-O2 -arch x86_64'
         ;;
     *BSD)
         THE_OS="BSD"
